src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java

```diff
@@ -51,4 +55,62 @@ private SpotlightBatch getSpotlightBatch(UUID spotlightBatchId) {
                 () -> new NotFoundException("A spotlight batch with id " + spotlightBatchId + " could not be found"));
     }
 
+    private GetSpotlightBatchErrorCountDTO orderSpotlightErrorStatusesByPriority(
+            List<SpotlightSubmission> filteredSubmissions) {
+        int apiErrorCount = 0;
+        int ggisErrorCount = 0;
+        int validationErrorCount = 0;
+
+        for (SpotlightSubmission submission : filteredSubmissions) {
+            switch (SpotlightSubmissionStatus.valueOf(submission.getStatus())) {
+                case SEND_ERROR:
+                    apiErrorCount++;
+                    break;
+                case GGIS_ERROR:
+                    ggisErrorCount++;
+                    break;
+                case VALIDATION_ERROR:
+                    validationErrorCount++;
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (apiErrorCount == 0 && ggisErrorCount == 0 && validationErrorCount == 0) {
+            return GetSpotlightBatchErrorCountDTO.builder().errorCount(0).errorStatus("OK").errorFound(false).build();
+        }
+
+        // Priority order: API > GGIS > VALIDATION. Validation is the lowest/default
+        // priority
+        // and will be overwritten if any higher-priority statuses exist.
+        int errorCount = validationErrorCount;
+        String errorStatus = "VALIDATION";
+
+        if (apiErrorCount > 0) {
+            errorCount = apiErrorCount;
+            errorStatus = "API";
+        }
+        else if (ggisErrorCount > 0) {
+            errorCount = ggisErrorCount;
+            errorStatus = "GGIS";
+        }
+        return GetSpotlightBatchErrorCountDTO.builder().errorCount(errorCount).errorStatus(errorStatus).errorFound(true)
+                .build();
+    }
+
+    public GetSpotlightBatchErrorCountDTO getSpotlightBatchErrorCount(Integer schemeId) {
+        // TODO: This will need to be refactored to get multiple batches if they all share
```

Comment:
Is this going to be actioned by another ticket currently in sprint?

Reply:
> Is this going to be actioned by another ticket currently in sprint?

I believe it's been actioned